GVFS/GVFS.UnitTests/Common/NuGetUpgrade/OrgNuGetUpgraderTests.cs

```diff
@@ -0,0 +1,175 @@
+                .WithOrgInfoServerUrl()
+                .Build();
+
+            this.upgraderConfig = new OrgNuGetUpgrader.OrgNuGetUpgraderConfig(this.tracer, mockGvfsConfig);
+            this.upgraderConfig.TryLoad(out _);
+
+            this.tracer = new MockTracer();
+
+            this.mockNuGetFeed = new Mock<NuGetFeed>(
+                DefaultUpgradeFeedUrl,
+                DefaultUpgradeFeedPackageName,
+                this.downloadDirectoryPath,
+                null,
+                this.tracer);
+
+            this.mockFileSystem = new MockFileSystem(
+                new MockDirectory(
+                    Path.GetDirectoryName(this.downloadDirectoryPath),
+                    new[] { new MockDirectory(this.downloadDirectoryPath, null, null) },
+                    null));
+
+            this.mockCredentialManager = new Mock<ICredentialStore>();
+            string credentialManagerString = "value";
+            string emptyString = string.Empty;
+            this.mockCredentialManager.Setup(foo => foo.TryGetCredential(It.IsAny<ITracer>(), It.IsAny<string>(), out credentialManagerString, out credentialManagerString, out credentialManagerString)).Returns(true);
+
+            this.httpMessageHandlerMock = new Mock<HttpMessageHandler>();
+
+            this.httpMessageHandlerMock.Protected().As<IHttpMessageHandlerProtectedMembers>()
+                .Setup(m => m.SendAsync(It.IsAny<HttpRequestMessage>(), It.IsAny<CancellationToken>()))
+                .ReturnsAsync(new HttpResponseMessage()
+                {
+                    StatusCode = HttpStatusCode.OK,
+                    Content = new StringContent(this.ConstructResponseContent(NewerVersion))
+                });
+
+            HttpClient httpClient = new HttpClient(this.httpMessageHandlerMock.Object);
+
+            this.upgrader = new OrgNuGetUpgrader(
+                CurrentVersion,
+                this.tracer,
+                this.mockFileSystem,
+                httpClient,
+                false,
+                false,
+                this.upgraderConfig,
+                "windows",
+                this.mockNuGetFeed.Object,
+                this.mockCredentialManager.Object);
+        }
+
+        [TestCase]
+        public void SupportsAnonymousQuery()
+        {
+            this.upgrader.SupportsAnonymousVersionQuery.ShouldBeTrue();
+        }
+
+        [TestCase]
+        public void TryQueryNewestVersion()
+        {
+            Version newVersion;
+            string message;
+
+            this.httpMessageHandlerMock.Protected().As<IHttpMessageHandlerProtectedMembers>()
+                .Setup(m => m.SendAsync(It.IsAny<HttpRequestMessage>(), It.IsAny<CancellationToken>()))
+                .ReturnsAsync(new HttpResponseMessage()
+                {
+                    StatusCode = HttpStatusCode.OK,
+                    Content = new StringContent(this.ConstructResponseContent(NewerVersion))
+                });

```

Comment:
Why do we need to set this up the exact same as in the Setup?  Can we remove one?

Reply:
Yes - thanks!